### PR TITLE
feat: Developer Guide - GUI - DataGrip

### PR DESCRIPTION
**Pull request by Icemap**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: develop/dev-guide-gui-datagrip.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+    ![DataGrip select data source](/media/develop/datagrip-data-source-select.jpg)
+
+7. Copy the JDBC string from the TiDB Cloud connection dialog and replace the `<your_password>` placeholder with your actual password. Then, paste it into `URL`, and the remaining parameters will be auto-filled. The example result is as follows:
+
+    ![DataGrip URL paste](/media/develop/datagrip-url-paste.jpg)
+
+8. Click **Download** missing driver files. If you have used DataGrip before, you can skip this step.
+
+9. Click **Test Connection** to establish a connection to TiDB Serverless Tier cluster.
```
Comment (sykp241095) [inline develop/dev-guide-gui-datagrip.md:88]: Don't need to input CA path in DataGrip for tidb serverless? @Icemap

Comment (Icemap) [reply develop/dev-guide-gui-datagrip.md:88]: Yep, the JDK uses the system's default CA path to link to the Serverless Tier.

**Review by sykp241095 — APPROVED** [review]

**Review by shizn — APPROVED** [review]

**Comment by Oreoxmt** [comment]
@Icemap https://github.com/Icemap/docs/pull/4 PTAL

**Comment by Icemap** [comment]
> @Icemap [Icemap#4](https://github.com/Icemap/docs/pull/4) PTAL

Done

File: develop/dev-guide-gui-datagrip.md
Comment (Oreoxmt) [inline develop/dev-guide-gui-datagrip.md:1]: preview: https://pingcap-docs-website-scaffold-git-preview-pingca-1c6c01-oreoxmt.vercel.app/tidb/dev/dev-guide-gui-datagrip

Comment (Icemap) [reply develop/dev-guide-gui-datagrip.md:1]: Looks good. Let me add a line to the third party summary page.

Comment (Icemap) [reply develop/dev-guide-gui-datagrip.md:1]: Oh wait, I needn't do that, cause the DataGrip is not the `Full` level support.

Comment (Icemap) [reply develop/dev-guide-gui-datagrip.md:1]: All good, IMO we can merge it now.

File: develop/dev-guide-gui-datagrip.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+5. Launch DataGrip and create a project to manage your connections.
+
+    ![Create a project in DataGrip](/media/develop/datagrip-create-project.jpg)
+
+6. In the newly created project, click **+** in the upper-left corner of the **Database Explorer** panel, and select **Data Source** > **Other** > **TiDB**.
+
+    ![Select a data source in DataGrip](/media/develop/datagrip-data-source-select.jpg)
+
+7. Copy the JDBC string from the TiDB Cloud connection dialog and replace the `<your_password>` placeholder with your actual password. Then, paste it into the **URL** field, and the remaining parameters will be auto-populate. An example result is as follows:
```
Comment (ran-huang) [inline develop/dev-guide-gui-datagrip.md:76]:
```suggestion
7. Copy the JDBC string from the TiDB Cloud connection dialog and replace `<your_password>`  with your actual password. Then, paste it into the **URL** field, and the remaining parameters will be auto-populated. An example result is as follows:
```

**Review by ran-huang — APPROVED** [review]

**Review by Oreoxmt — APPROVED** [review]

**Pull request merged**